### ISSUE TITLE
[1.15.2] Fruit Trees too common and ignoring config

### ISSUE THREAD

**Issue by Besiegel** [MatrexsVigil/phc2trees#2]
Fruit trees seem to spawn literally everywhere they have room, and seem to be ignoring the config to decrease spawn rate by increasing numbers. It appears that everytime it is checking if it can place a tree, and the game returns yes, it does, even though the spawnrate in the config should result in a "no". This results in fruit trees dominating the landscape.

**Comment by curlywaffles** [comment]
Can confirm this. Tried maxing out the number in the config to no avail.

**Comment by IchigoGames** [comment]
if turn them all down from 500 to 50 and it seem to be a good number not to much

**Comment by KrinonValian** [comment]
Changed spawnrate from 500 to 20 and still having massive amounts of fruit trees spawning.

**Comment by IchigoGames** [comment]
ya i just made a new world still set to 50% there are still alot of trees must of just been in a biome where they was not as meany before.

**Comment by GatorThing** [comment]
I have also been observing this behavior. I have made 2 worlds with the same seed, teleported to the same location and looked the same direction for the following screenshots, the only difference is one has the spawn probability config set to 1 and the other 1000000000, and theres almost 0 difference between them as you can see. Each of the brighter green trees are fruit trees and with the worlds being at the extremes of the config i would expect one picture would have almost no fruit trees.

![2020-04-17_16 42 35](https://user-images.githubusercontent.com/22823400/79638800-70080b00-8155-11ea-9b62-072d2de4ee2e.png)
![2020-04-17_17 02 09](https://user-images.githubusercontent.com/22823400/79638806-76968280-8155-11ea-8cbf-bf66d926cc64.png)

**Comment by IchigoGames** [comment]
this could be closed seems to be fixed now

**Comment by Linguardium** [comment]
seems to still occur on building but may have been corrected in closed source? (didnt test release)
I ran into this issue when building my own tree mod. Each tree is added as an individual feature, and each tree uses the spawn chance (1/chance) to see if it is allowed to spawn. 

This means that even at 1/100,000 chance, that just means that each tree type has 1/100,00 chance. so the chance of ANY spawn isnt really 1/X, but the chance of any 1 tree type spawning in each attempt is 1/X (a much higher chance)
